apps/server/src/app/fhir.controller.ts

```diff
@@ -95,11 +96,81 @@ export class FhirController extends BaseController {
       method: 'DELETE'
     };
 
-    this.logger.log('Sending PUT request to FHIR server with the new resource ID');
-
     // Create the new resource with the new id
     await this.httpService.request(createOptions).toPromise();
 
+    const searchForReference = (searchResourceType: string, searchParameter: string) => {
+
+      return new Promise(async (resolve) => {
+        const params = {};    // DONT use _summary=true. The results may end up getting used to update the resource.
+        params[searchParameter] = `${resourceType}/${currentId}`;
+        const searchUrl = buildUrl(fhirServerBase, searchResourceType, null, null, params);
+
+        const results = await this.httpService.get(searchUrl).toPromise();
+        const bundle: Bundle = results.data;
+        const resources = (bundle.entry || []).map(entry => <DomainResource> entry.resource);
+
+        resolve(resources);
+      });
+    };
+
+    const searchPromises = [];
+    // These search parameters apply to both STU3 and R4 servers
+    searchPromises.push(searchForReference('ImplementationGuide', 'resource'));
+
+    const allResults = await Promise.all(searchPromises);
+
+    const allResources = allResults.reduce((prev, curr) => {
+      return prev.concat(curr);
+    }, []);
+
+    const findReference = (obj: any) => {
+      if (!obj) return;
+
+      if (obj.reference === `${resourceType}/${currentId}`) {
+        return obj;
+      }
+
+      const propertyNames = Object.keys(obj);
+
+      for (let i = 0; i < propertyNames.length; i++) {
+        const propertyName = propertyNames[i];
+
+        if (typeof obj[propertyName] === 'object') {
+          const foundReference = findReference(obj[propertyName]);
+
+          if (foundReference) {
+            return foundReference;
+          }
+        }
+      }
+    };
+
+    allResults.forEach(resource => {
+      const foundReference = findReference(resource); 
+      foundReference.reference =  `${resourceType}/${newId}`;
```

Comment:
I think null/undefined checking is needed here. `findReference()` can return undefined, and if it does, this will try to set a sub-property of an undefined object, producing a NullPointerException.